### PR TITLE
Bump bugfix: Wallet connect disable connection backdrop and add tooltip for amount field

### PR DESCRIPTION
**Pull request by TejasvCypherock**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [CSI-4107 Bugfix: update amount field, fix Api errors, disable connection backdrop](https://app.clickup.com/t/9002019994/CSI-4107)

File: cysync/src/pages/mainApp/sidebar/wallet/send/formStepComponents/Recipient.tsx [hunk 1]
```diff
@@ -804,36 +804,50 @@ const Recipient: React.FC<StepComponentProps> = props => {
...
-                disabled={!!txnParams?.value}
-              >
-                Send Max
-              </Button>
+          <Tooltip
+            title={
+              txnParams?.value !== undefined &&
+              batchRecipientData[0].amount === '0'
+                ? "The amount being 0 doesn't necessarily mean that no amount will be deducted.\nContract data in this transaction might contain the actual amount."
+                : ''
```
Comment (irshadCypherock) [inline cysync/src/pages/mainApp/sidebar/wallet/send/formStepComponents/Recipient.tsx:812]: Also check if there is contract data present, and finalize the tooltip message

File: cysync/src/pages/mainApp/sidebar/wallet/send/stepperForm.tsx [hunk 2]
```diff
@@ -387,14 +387,15 @@ const SendForm: React.FC<StepperProps> = ({
...
-    );
+    const estimatedLimit =
+      (await sendTransaction.handleEstimateGasLimit(
+        fromAddress,
+        toAddress,
+        network,
+        amount,
+        contractAddress,
+        txnParams?.data
+      )) ?? parseInt(txnParams?.gas, 16);
```
Comment (irshadCypherock) [inline cysync/src/pages/mainApp/sidebar/wallet/send/stepperForm.tsx:398]: Add a check for undefined gas

File: cysync/src/pages/mainApp/sidebar/wallet/send/stepperForm.tsx [hunk 3]
```diff
@@ -554,10 +555,13 @@ const SendForm: React.FC<StepperProps> = ({
...
-        if (!(amount.isZero() && isEthereum) && !maxSend) {
-          error = 'Enter a valid amount.';
-          isValid = false;
+        // Allow '0' amount for wallet connect transactions
+        if (!(amount.isZero() && txnParams?.data)) {
+          // Allow `0` amount transaction on ETH, and 0 amount is valid when it's a max send txn
+          if (!(amount.isZero() && isEthereum) && !maxSend) {
+            error = 'Enter a valid amount.';
+            isValid = false;
+          }
```
Comment (irshadCypherock) [inline cysync/src/pages/mainApp/sidebar/wallet/send/stepperForm.tsx:564]: Allow `0` value txn on ETH

File: cysync/src/store/provider/walletConnectProvider/index.tsx [hunk 4]
```diff
@@ -221,6 +221,7 @@ export const WalletConnectProvider: React.FC = ({ children }) => {
       payload?.params &&
       ACCEPTED_CALL_METHODS.includes(payload?.method)
     ) {
+      ipcRenderer.send('focus');
```
Comment (irshadCypherock) [inline cysync/src/store/provider/walletConnectProvider/index.tsx:224]: Create a handler for this event in main process

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**